doc/source/whatsnew/v0.23.0.txt

```diff
@@ -294,6 +295,7 @@ Conversion
 - Bug in :meth:`DatetimeIndex.astype` when converting between timezone aware dtypes, and converting from timezone aware to naive (:issue:`18951`)
 - Bug in :class:`FY5253` where ``datetime`` addition and subtraction incremented incorrectly for dates on the year-end but not normalized to midnight (:issue:`18854`)
 - Bug in :class:`DatetimeIndex` where adding or subtracting an array-like of ``DateOffset`` objects either raised (``np.array``, ``pd.Index``) or broadcast incorrectly (``pd.Series``) (:issue:`18849`)
+- Bug in ``IntervalDtype`` when constructing two instances with subtype ``CategoricalDtype`` where the second instance used cached attributes from the first (:issue:`18980`)
```

Comment:
are you mixing IntervalDtype and CategoricalDtype here?

Reply:
I'm referring to the behavior where you can only store one `CategoricalDtype` in the subtype cache:
```python
In [12]: idt1 = IntervalDtype(CategoricalDtype(list('abc'), True))

In [13]: idt2 = IntervalDtype(CategoricalDtype(list('wxyz'), False))

In [14]: idt2.subtype
Out[14]: CategoricalDtype(categories=['a', 'b', 'c'], ordered=True)
```
Once you've created an `IntervalDtype` with `CategoricalDtype` subtype, you can't create any other unique ones.  Since the caching was previously based on `str(subtype)`, and `str(CategoricalDtype(*))` always returns `'category'`, trying to create secondary instances, even with different `categories` or `ordered`, will always hit the cache and pull out the first one.